grid-client/calculator/calculate.go

```diff
@@ -253,29 +271,54 @@ func (c *Calculator) calculateTotalContractsOverdueOnNode(nodeID uint32, allowan
 	if err != nil {
 		return 0, errors.Wrapf(err, "failed to get contracts for node ID %d", nodeID)
 	}
+
+	var wg sync.WaitGroup
+	var mu sync.Mutex
+	wg.Add(len(contracts))
+
 	var totalCost int64 = 0
+	errList := make([]error, 0)
```

Comment:
you can use "github.com/hashicorp/go-multierror" to accumilate the error instead of a slice of errors